harness/determined/searcher/_search_method.py

```diff
@@ -76,12 +92,19 @@ def _from_bindings(
 
 
 class Operation:
+    """
+    Abstract base class for all Operations
+    """
     @abstractmethod
     def _to_searcher_operation(self) -> bindings.v1SearcherOperation:
         pass
 
 
 class ValidateAfter(Operation):
+    """
+    Operation signaling the trial to train until its total batches trained
+    equals the specified length
```

Comment:
Isn't it either epochs or batches (depending on what is passed as `unit` in the yaml config)?